posts/2024-06-07_admiral_1.1_is.../admiral_1.1_is_here.qmd

```diff
@@ -0,0 +1,248 @@
+  ~USUBJID, ~AESTDTC,           ~AESEQ, ~AEDECOD,
+  "01",     "2021-01-03T10:56", 1,      "Flu",
+  "01",     "2021-03-04",       2,      "Cough",
+  "01",     "2021",             3,      "Flu"
+) %>%
+  mutate(STUDYID = "AB42")
+
+ae_ext <- derive_vars_dt(
+  ae,
+  dtc = AESTDTC,
+  new_vars_prefix = "AEST",
+  highest_imputation = "M",
+  flag_imputation = "none"
+)
+
+ttae <- event_source(
+  dataset_name = "ae",
+  date = AESTDT,
+  set_values_to = exprs(
+    EVNTDESC = "AE",
+    SRCDOM = "AE",
+    SRCVAR = "AESTDTC",
+    SRCSEQ = AESEQ
+  )
+)
+
+eos <- censor_source(
+  dataset_name = "adsl",
+  date = EOSDT,
+  set_values_to = exprs(
+    EVNTDESC = "END OF STUDY",
+    SRCDOM = "ADSL",
+    SRCVAR = "EOSDT"
+  )
+)
+
+derive_param_tte(
+  dataset_adsl = adsl,
+  by_vars = exprs(AEDECOD),
+  start_date = TRTSDT,
+  event_conditions = list(ttae),
+  censor_conditions = list(eos),
+  source_datasets = list(adsl = adsl, ae = ae_ext),
+  set_values_to = exprs(
+    PARAMCD = "TTAE", # This is incorrect syntax! Not unique!
+    PARAM = paste("Time to First", AEDECOD, "Adverse Event"),
+    PARCAT1 = "TTAE",
+    PARCAT2 = AEDECOD
+  )
+)
+```
+
+```{r, fig.align = 'center', echo=FALSE}
+knitr::include_graphics("admiral_error.png")
+```
+
+While we hope you don't encounter too many errors and warnings while working with {admiral}, we also hope that when you do, our improved messaging will put you on the right path towards correcting them 😀.
+
+# New metadata object for ISO 3166 country codes
+
+{admiral} now offers a new metadata object, admiral::country_code_lookup(), that can aid metadata added to decode countries based on [ISO 3166 codes](https://www.iso.org/iso-3166-country-codes.html).
+Our hugely-flexible admiral::derive_vars_merged() then comes in handy if you ever need to leverage this metadata:
+
+```{r}
+adsl <- tribble(
+  ~USUBJID, ~COUNTRY,
+  "ST01-01", "AUT",
+  "ST01-02", "MWI",
+  "ST01-03", "GBR",
+  "ST01-04", "CHE",
+  "ST01-05", "NOR",
+  "ST01-06", "JPN",
+  "ST01-07", "USA"
+)
+
+adsl %>%
+  derive_vars_merged(
+    dataset_add = country_code_lookup,
+    new_vars = exprs(COUNTRYN = country_number, COUNTRYL = country_name),
+    by_vars = exprs(COUNTRY = country_code)
+  )
+```
+
+If you were worried about having to type these up manually yourself, fear not as we now have you covered!
+
+# New helper functions for metadata-driven programming
+
+Driven by a request from our GSK users, the {admiral} back-end now contains helper functions to admiral::derive_vars_merged() (admiral::get_flagged_records()) and admiral::derive_vars_query() (admiral::get_vars_query()) so that those can be called independently if required.
+This should aid efforts to automate execution of {admiral} code in a metadata-drive way.
+You can read more about the the reasoning behind this update in the following [discussion](https://github.com/pharmaverse/admiral/discussions/2349) on GitHub.
+
+# Enhancements to admiral::derive_var_trtemfl()
```

Comment:
```suggestion
# Enhancements to `derive_var_trtemfl()`
```

The section title breaks into another line -
![image](https://github.com/pharmaverse/blog/assets/47867131/81201d05-e37f-416f-bf62-6bba2347f37f)